test/noise_function.jl

```diff
@@ -1,33 +1,32 @@
 @testset "NoiseFunction" begin
 
-using DiffEqNoiseProcess, DiffEqBase, Test
-f = (u,p,t) -> exp(t)
+  using DiffEqNoiseProcess, DiffEqBase, Test
+  f = (u,p,t) -> exp(t)
 
-W = NoiseFunction(0.0,f)
+  W = NoiseFunction(0.0,f)
 
-dt = 0.1
-calculate_step!(W,dt,nothing,nothing)
+  dt = 0.1
+  calculate_step!(W,dt,nothing,nothing)
 
-for i in 1:10
-  accept_step!(W,dt,nothing,nothing)
-end
-
-@test W(W.curt + W.dt)[1] - W.curW == W.dW
+  for i in 1:10
+    accept_step!(W,dt,nothing,nothing)
+  end
 
-prob = NoiseProblem(W,(0.0,1.0));
-sol = solve(prob;dt=0.1)
+  @test W(W.curt + W.dt)[1] - W.curW == W.dW
 
-@test sol(W.curt + W.dt)[1] - W.curW == W.dW
+  prob = NoiseProblem(W,(0.0,1.0));
+  sol = solve(prob;dt=0.1)
 
-f = (out,u,p,t) -> (out.=exp(t))
-W = NoiseFunction(0.0,f,noise_prototype=rand(4))
-prob = NoiseProblem(W,(0.0,1.0))
-sol = solve(prob;dt=0.1)
+  @test sol(W.curt + W.dt)[1] - W.curW == W.dW
 
-out1 = rand(4)
-out2 = nothing
-sol(out1,out2,nothing,nothing,0.5)
+  f = (out,u,p,t) -> (out.=exp(t))
+  W = NoiseFunction(0.0,f,noise_prototype=rand(4))
+  prob = NoiseProblem(W,(0.0,1.0));
+  sol = solve(prob;dt=0.1)
 
-@test out1 == sol(0.5)[1]
+  out1 = rand(4)
+  out2 = nothing
+  sol(out1,out2,nothing,nothing,0.5)
 
+  @test out1 == sol(0.5)[1]
```

Comment:
It might be good to test the memory size of the solution object somehow?

Reply:
Yeah, you mean besides the tests that test the cache sizes of the tree?
```julia
@test length(W.seeds) == 2^(W.tree_depth)
@test length(W.W) == 2^(W.tree_depth)+1
@test length(W.t) == 2^(W.tree_depth)+1
@test length(unique(W.seeds)) == length(W.seeds)
```
do we have good examples somewhere?

Reply:
Yes, on the low storage RK methods we use Base.summarysize to get the memory size. In that case, we know that there should be a total of 4 vectors saved in there (for this specific one, if implemented in the optimal memory way), so we divide by the size of `u0` to show that indeed the total sizeof the integrator is 4*u0 memory.

https://github.com/SciML/OrdinaryDiffEq.jl/blob/dadfdd63b7ab267fc15bd6a1ed7898ad8f890bdf/test/algconvergence/ode_ssprk_tests.jl#L74-L78

We can probably do something similar here. Take u0 large enough and show that integer dividing by its memory size gives 0, since the storage of the VBT should just be a tree of seeds which is of size much less than a single u0, IIUC

Reply:
Well actually, it should be 2*size(W), since there looks to be 2 cache variables for making it non-allocating? So the goal would be to show that those two cache variables are all that are ever used in this object.

Reply:
I added a check in the "add memory test " commit.